File: wpilibc/src/main/native/cpp/Commands/PrintCommand.cpp
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2011-2018 FIRST. All Rights Reserved.                        */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

#include "Commands/PrintCommand.h"

#include <llvm/raw_ostream.h>

using namespace frc;

PrintCommand::PrintCommand(const llvm::Twine& message)
    : InstantCommand("Print \"" + message + llvm::Twine('"')) {
  m_message = message.str();
}

void PrintCommand::Initialize() { llvm::outs() << m_message << '\n'; }

File: wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/PrintCommand.java
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

package edu.wpi.first.wpilibj2.command;

/**
 * A command that prints a string when initialized.
 */
public class PrintCommand extends InstantCommand {
  /**
   * Creates a new a PrintCommand.
   *
   * @param message the message to print
   */
  public PrintCommand(String message) {
    super(() -> System.out.println(message));
  }

  @Override
  public boolean runsWhenDisabled() {
    return true;
  }
}
